financial-templates-lib/BlockNumberFinder.js

```diff
@@ -0,0 +1,64 @@
+module.exports = async (web3, targetTimestamp, lowerLimitStamp, higherLimitStamp) => {
```

Comment:
Drive-by high level comments:
1. Can you give a brief explanation (in comments or in this thread) on what these arguments do?
2. Will this work for the price feed? My intuition is that it won't in its current form because the price feed needs to start querying at the latest block that is 1) before timestamp x (which this PR seems to accomplish) AND 2) contains a `Sync` event from the Uniswap contract. IIUC, this PR will not accomplish the latter, right?

Reply:
1. This is still WIP and I will add detailed comments and instructions on how to use it when it moves out of draft.
2. The goal is to get it to work for the price feed as well as the monitors. Specifically when querying _events_. However your `AND 2)` restriction will not work if there has not been an even in the last period. I am thinking about how to overcome this to work with the UniSwapPriceFeed and will push some ideas when I circle back to this PR.